internal/resolution/variables/installed_package.go

```diff
@@ -0,0 +1,32 @@
+package variables
+
+import (
+	"fmt"
+
+	"github.com/operator-framework/deppy/pkg/deppy"
+	"github.com/operator-framework/deppy/pkg/deppy/constraint"
+	"github.com/operator-framework/deppy/pkg/deppy/input"
+
+	olmentity "github.com/operator-framework/operator-controller/internal/resolution/entities"
+)
+
+type InstalledPackageVariable struct {
+	*input.SimpleVariable
+	bundleEntities []*olmentity.BundleEntity
+}
+
+func (r *InstalledPackageVariable) BundleEntities() []*olmentity.BundleEntity {
+	return r.bundleEntities
+}
+
+func NewInstalledPackageVariable(bundleImage string, bundleEntities []*olmentity.BundleEntity) *InstalledPackageVariable {
+	id := deppy.IdentifierFromString(fmt.Sprintf("installed package %s", bundleImage))
+	entityIDs := make([]deppy.Identifier, 0, len(bundleEntities))
+	for _, bundle := range bundleEntities {
+		entityIDs = append(entityIDs, bundle.ID)
+	}
+	return &InstalledPackageVariable{
+		SimpleVariable: input.NewSimpleVariable(id, constraint.Mandatory(), constraint.Dependency(entityIDs...)),
+		bundleEntities: bundleEntities,
+	}
+}
```

Comment:
@jmprusi what is the difference between this and `RequiredPackageVariable`?

Reply:
They are semantically different concepts that get translated to the same sort of structure (variable with mandatory and dependency constraints). I guess we could separate that out as a concept if we want to simplify the code here a bit. wdyt?

Reply:
> They are semantically different concepts that get translated to the same sort of structure (variable with mandatory and dependency constraints).

@perdasilva Is it just semantic difference? Or are they/will they be processed differently? As far as I understnad - they are exactly the same today.

> I guess we could separate that out as a concept if we want to simplify the code here a bit. wdyt?

For now I'm just trying to understand how it works. But I don't follow - what do you mean by separating? They are already separate structures.

Reply:
For now it is. It could be that these two evolve on different paths. I meant separate out the pattern of a Variable with a Mandatory and Dependency constraints. I think it could be a common pattern. Then we can just typedef the required and installed package variables as that.

Reply:
Thanks for the clarification.

> I meant separate out the pattern of a Variable with a Mandatory and Dependency constraints. I think it could be a common pattern. Then we can just typedef the required and installed package variables as that.

Sounds like a good idea.